MovieDBApp/MovieDBApp/Scenes/DetailMovieScreen/MovieDetailViewModel.swift

```diff
@@ -12,38 +12,63 @@ import RxCocoa
 struct MovieDetailViewModel {
     let useCase: MovieDetailUseCaseType
     let navigator: MovieDetailNavigatorType
-    let movieId: Int
+    let movie: Movie
     
     struct Input {
         let loadTrigger: Driver<Void>
-        let selectMovieTrigger: Driver<Int>
+        let selectMovieTrigger: Driver<Movie>
+        let likeMovieTrigger: Driver<Bool>
     }
     
     struct Output {
         let movieDetail: Driver<[MovieDetailSectionModel]>
-        let selectedMovieId: Driver<Int>
+        let selectedMovieId: Driver<Movie>
+        let handleStatus: [Driver<Bool>]
     }
     
     func transform(input: Input) -> Output {
+        let handleLike = BehaviorRelay<Bool>(value: false)
+        
+        let isFavoriteMovie = input.loadTrigger
+            .flatMapLatest { _ in
+                return useCase.checkMovieExist(id: movie.id)
+                    .asDriver(onErrorDriveWith: .empty())
+            }
+            .do(onNext: handleLike.accept(_:))
+        
+        let likeFavoriteMovie = input.likeMovieTrigger
+            .flatMapLatest { status in
+                return status ?
+                    useCase.deleteFavoriteMovie(id: movie.id).asDriver(onErrorDriveWith: .empty()) :
+                    useCase.addFavoriteMovie(movie: movie).asDriver(onErrorDriveWith: .empty())
```

Comment:
sao lại xóa rồi add là sao e?

Reply:
dạ cái này e toán tử 3 ngôi với biến status ở trên ạ, biến status giúp e check là cái movie đó trước khi click nó là true hay false, nếu là true thì có nghĩ nó có trong favorite rồi nên e gọi delete còn nếu false là chưa có e gọi addFavorite ạ

Reply:
```suggestion
                    ? useCase.deleteFavoriteMovie(id: movie.id).asDriver(onErrorDriveWith: .empty())
                    : useCase.addFavoriteMovie(movie: movie).asDriver(onErrorDriveWith: .empty())
```